src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -5342,6 +5342,215 @@ internal override AstVisitAction InternalVisit(AstVisitor visitor)
+            }
+
+            LhsStatement = lhsStatement;
+            RhsStatement = rhsStatement;
+            Operator = chainOperator;
+
+            SetParent(LhsStatement);
+            SetParent(RhsStatement);
+        }
+
+        /// <summary>
+        /// The left hand statement in the chain.
+        /// </summary>
+        public PipelineBaseAst LhsStatement { get; }
+
+        /// <summary>
+        /// The right hand statement in the chain.
+        /// </summary>
+        public StatementAst RhsStatement { get; }
+
+        /// <summary>
+        /// The chaining operator used.
+        /// </summary>
+        public TokenKind Operator { get; }
+
+        /// <summary>
+        /// Create a copy of this Ast.
+        /// </summary>
+        public override Ast Copy()
+        {
+            return new StatementChainAst(Extent, CopyElement(LhsStatement), CopyElement(RhsStatement), Operator);
+        }
+
+        internal override object Accept(ICustomAstVisitor visitor)
+        {
+            if (!(visitor is ICustomAstVisitor2 visitor2))
+            {
+                return null;
+            }
+
+            return visitor2.VisitStatementChain(this);
+        }
+
+        internal override AstVisitAction InternalVisit(AstVisitor visitor)
+        {
+            var visitAction = AstVisitAction.Continue;
+            if (visitor is AstVisitor2 visitor2)
+            {
+                visitAction = visitor2.VisitStatementChain(this);
+
+                if (visitAction != AstVisitAction.Continue)
+                {
+                    return visitAction;
+                }
+            }
+
+            visitAction = LhsStatement.InternalVisit(visitor);
+
+            if (visitAction == AstVisitAction.StopVisit)
+            {
+                return visitAction;
+            }
+
+            return RhsStatement.InternalVisit(visitor);
+        }
+    }
+
+    /// <summary>
+    /// A command-oriented flow-controlled pipeline chain.
+    /// E.g. <c>npm build &amp;&amp; npm test</c> or <c>Get-Content -Raw ./file.txt || "default"</c>.
+    /// </summary>
+    public class PipelineChainAst : PipelineBaseAst
+    {
+        /// <summary>
+        /// Create a new statement chain AST from two statements and an operator.
+        /// </summary>
+        /// <param name="extent">The extent of the chained statement.</param>
+        /// <param name="lhsPipeline">The statement to the left of the operator.</param>
+        /// <param name="rhsPipeline">The statement to the right of the operator.</param>
+        /// <param name="chainOperator">The operator used.</param>
+        /// <param name="background">If true, the pipeline chain should be backgrounded</param>
+        public PipelineChainAst(
+            IScriptExtent extent,
+            PipelineBaseAst lhsPipeline,
+            PipelineBaseAst rhsPipeline,
+            TokenKind chainOperator,
+            bool background = false)
```

Comment:
So the `PipelineChainAst` supports background operator but `StatementChainAst` doesn't.
Two questions:
1. Now we will support `a || b || c & &`? (the first `&` goes with `c` and the second `&` goes with the `PipelineChainAst`)
    It feels weird looking at this ...
2. Would the error message be clear for script like `a || b || return &`? (thinking about `unexpected token &` vs more descriptive one that explains why `&` is not supported here).

General question, should `PipelineChainAst` support `&` operator ... I know this should be the question answered in RFC, but but ...

Reply:
For visibility, here are some related thoughts about [assigning the results of a pipeline chain](https://github.com/PowerShell/PowerShell-RFC/pull/192#discussion_r293176686) and about [pipeline vs statement chaining with the background operator](https://github.com/PowerShell/PowerShell-RFC/pull/192#discussion_r293407287).

Reply:
> So the PipelineChainAst supports background operator but StatementChainAst doesn't.

Indeed, just as `PipelineAst` does but `StatementAst` does not.

> Now we will support a || b || c & &?

No, `PipelineRule` looks for the `&` still but `PipelineChainRule` decides to stop parsing when it's seen.

Reply:
After discussing with the @PowerShell/powershell-committee, I've taken out the "statement chain" concept, resolving the strangeness here